ambry-store/src/main/java/com/github/ambry/store/HybridCompactionPolicy.java

```diff
@@ -0,0 +1,194 @@
+  }
+
+  /**
+   * Recover the {@link CompactionPolicySwitchInfo} from backup file.
+   * {
+   *   "compactionPolicyCounter" : {
+   *     "storeCompactionPolicySwitchPeriod" : 3,
+   *     "counter" : 1,
+   *     "value" : 1
+   *   },
+   *   "lastCompactAllTime" : 1593463435900
+   * }
+   * @param file the backup file stores {@link CompactionPolicySwitchInfo}
+   */
+  private CompactionPolicySwitchInfo recoverCompactionPolicySwitchInfo(File file) {
+    try {
+      ObjectMapper objectMapper = new ObjectMapper();
+      return objectMapper.readValue(file, CompactionPolicySwitchInfo.class);
+    } catch (IOException e) {
+      logger.error("Could not deserialize file : {} into {} Object", file, CompactionPolicySwitchInfo.class.getName());
+      return new CompactionPolicySwitchInfo(new CompactionPolicyCounter(storeConfig.storeCompactionPolicySwitchPeriod),
+          INIT_COMPACT_ALL_TIME);
+    }
+  }
+
+  /**
+   * Selects which compaction policy to use for current compaction cycle.
+   * @return {@link CompactionPolicy} to use for current compaction cycle(Selects between {@link StatsBasedCompactionPolicy} and {@link CompactAllPolicy}).
+   */
+  CompactionPolicy selectCompactionPolicyAndUpdateCompactionPolicySwitchInfo(
+      CompactionPolicySwitchInfo compactionPolicySwitchInfo) {
+    if (compactionPolicySwitchInfo == null) {
+      logger.trace("CompactionPolicySwitchInfo is null");
+      return new StatsBasedCompactionPolicy(storeConfig, time);
+    }
+    if (readyToTriggerCompactionAllPolicy(compactionPolicySwitchInfo)) {
+      logger.trace("Return CompactAllPolicy this round");
+      updateCompactionInfo(compactionPolicySwitchInfo);
+      return new CompactAllPolicy(storeConfig, time);
+    } else {
+      if (compactionPolicySwitchInfo.getCompactionPolicyCounter() == null) {
+        logger.trace("Counter is null");
+      } else {
+        logger.trace("Return StatsBasedCompactionPolicy this round");
+      }
+      return new StatsBasedCompactionPolicy(storeConfig, time);
+    }
+  }
+
+  /**
+   * Determine which compactionPolicy to use for current compaction cycle.
+   * @param compactionPolicySwitchInfo the info to determine which {@link CompactionPolicy} to use this round.
+   * @return {@code true} if the counter value equals to 0 or it's storeCompactionPolicySwitchPeriod days past the start time of CompactAllPolicy.
+   */
+  private boolean readyToTriggerCompactionAllPolicy(CompactionPolicySwitchInfo compactionPolicySwitchInfo) {
+    return compactionPolicySwitchInfo.getCompactionPolicyCounter() != null
+        && compactionPolicySwitchInfo.getCompactionPolicyCounter().getValue() == 0 ||
+        compactionPolicySwitchInfo.getLastCompactAllTime() + TimeUnit.DAYS.toMillis(
+            storeConfig.storeCompactionPolicySwitchPeriod) <= System.currentTimeMillis();
+  }
+
+  /**
+   * Update the {@link CompactionPolicySwitchInfo} before the start of {@link CompactAllPolicy}
```

Comment:
In this situation, it will recover the counter value in file which is 0 since we trigger compactAllPolicy before host crashes,  so which means after restart it will run compactAllPolicy one more time.